go-controller/pkg/util/multi_network.go

```diff
@@ -662,19 +662,52 @@ func GetPodNADToNetworkMapping(pod *kapi.Pod, nInfo NetInfo) (bool, map[string]*
 	for _, network := range allNetworks {
 		nadName := GetNADName(network.Namespace, network.Name)
 		if nInfo.HasNAD(nadName) {
+			if nInfo.IsPrimaryNetwork() {
+				return false, nil, fmt.Errorf("unexpected primary network %q specified with a NetworkSelectionElement %+v", nInfo.GetNetworkName(), network)
+			}
 			if _, ok := networkSelections[nadName]; ok {
 				return false, nil, fmt.Errorf("unexpected error: more than one of the same NAD %s specified for pod %s",
 					nadName, podDesc)
 			}
 			networkSelections[nadName] = network
 		}
 	}
+
 	if len(networkSelections) == 0 {
 		return false, nil, nil
 	}
+
 	return true, networkSelections, nil
 }
 
+func GetPodNADToNetworkMappingWithActiveNetwork(pod *kapi.Pod, nInfo NetInfo, activeNetwork NetInfo) (bool, map[string]*nettypes.NetworkSelectionElement, error) {
```

Comment:
What about `GetPodNetworkMappings` ? 

Do you need to call this function from the CNI side ? If not, we could even make it a network controller method, and grab the active network from within it. This would simplify the callers.

Reply:
> What about `GetPodNetworkMappings` ?
>

What do you mean ? 
 
> Do you need to call this function from the CNI side ? If not, we could even make it a network controller method, and grab the active network from within it. This would simplify the callers.

Since it's also needed by pod allocator we need this to be a free function is not only about CNI.

Reply:
> > What about `GetPodNetworkMappings` ?
> 
> What do you mean ?

I mean we should simplify the name of the func / method. And I'm proposing that particular fn name.

> 
> > Do you need to call this function from the CNI side ? If not, we could even make it a network controller method, and grab the active network from within it. This would simplify the callers.
> 
> Since it's also needed by pod allocator we need this to be a free function is not only about CNI.

Hm. Still, IMO, it would be preferable to have this grab the active network itself. This is a nit.

Reply:
> > > What about `GetPodNetworkMappings` ?
> > 
> > 
> > What do you mean ?
> 
> I mean we should simplify the name of the func / method. And I'm proposing that particular fn name.
> 

Let's be explicit here and be able to correlate this name with function it's caling inside.

> > > Do you need to call this function from the CNI side ? If not, we could even make it a network controller method, and grab the active network from within it. This would simplify the callers.
> > 
> > 
> > Since it's also needed by pod allocator we need this to be a free function is not only about CNI.
> 
> Hm. Still, IMO, it would be preferable to have this grab the active network itself. This is a nit.

I think that is making the function doing too much stuff, also we may want to "cache" the active network in the future at controllers since we cannot change it after configure it so just passing the active network from whatever is comming from makes more sense to me.

Reply:
your call.